src/jit/codegenxarch.cpp

```diff
@@ -5279,86 +5325,161 @@ void CodeGen::genConsumePutStructArgStk(
 }
 #endif // FEATURE_UNIX_AMD64_STRUCT_PASSING
 
-void CodeGen::genConsumeBlockOp(GenTreeBlkOp* blkNode, regNumber dstReg, regNumber srcReg, regNumber sizeReg)
+//------------------------------------------------------------------------
+// genConsumeBlockSize: Ensure that the block size is in the given register
+//
+// Arguments:
+//    blkNode - The block node
+//    sizeReg - The register into which the block's size should go
+//
+
+void CodeGen::genConsumeBlockSize(GenTreeBlk* blkNode, regNumber sizeReg)
```

Comment:
Considering the magnitude of changes, i would suggest to subject these changes to the following additional test runs (both on windows and Linux). Note that these could be triggered by typing in PR comment window.

- corefx baseline
- Jitstress2
- gcstress

I am assuming that you will also be measuring throughput numbers to make sure that there is no regression.